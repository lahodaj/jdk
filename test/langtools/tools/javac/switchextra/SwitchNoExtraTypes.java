/*
 * @test /nodynamiccopyright/
 * @bug 8206986
 * @summary Verify switch over boolean/long/float/double is not allowed.
 * @compile/fail/ref=SwitchNoExtraTypes.out -XDrawDiagnostics SwitchNoExtraTypes.java
 * @compile --enable-preview --source 22 SwitchNoExtraTypes.java
 */

public class SwitchNoExtraTypes {

    private void switchBoolean(boolean b) {
        switch (b) {
            case true: return ;
            default:
        }
    }

    private void switchLong(long l) {
        switch (l) {
            case 0l: return ;
            default:
        }
    }

    private void switchFloat(float f) {
        switch (f) {
            case 0f: return ;
            default:
        }
    }

    private void switchDouble(double d) {
        switch (d) {
            case 0d: return ;
            default:
        }
    }

}
