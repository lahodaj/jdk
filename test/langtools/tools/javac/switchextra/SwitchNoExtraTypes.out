SwitchNoExtraTypes.java:12:16: compiler.err.selector.type.not.allowed: boolean
SwitchNoExtraTypes.java:19:16: compiler.err.selector.type.not.allowed: long
SwitchNoExtraTypes.java:26:16: compiler.err.selector.type.not.allowed: float
SwitchNoExtraTypes.java:33:16: compiler.err.selector.type.not.allowed: double
4 errors
